common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -300,6 +275,63 @@ public Boolean run() {
         LINUX_OS_CLASSIFIERS = Collections.unmodifiableSet(availableClassifiers);
     }
 
+    private static Data getData() {
+        if (DATA == null) {
+            initialize();
+        }
+        return DATA;
+    }
+
+    private static final Object lock = new Object();
+    private static void initialize() {
+        synchronized (lock) {
+            if (DATA == null) {
+                final boolean useDirectBufferNoCleaner;
+                final AtomicLong directMemoryCounter;
+
+                final long maxDirectMemory0 = maxDirectMemory0();
+
+                // Here is how the system property is used:
+                //
+                // * <  0  - Don't use cleaner, and inherit max direct memory from java. In this case the
+                //           "practical max direct memory" would be 2 * max memory as defined by the JDK.
+                // * == 0  - Use cleaner, Netty will not enforce max memory, and instead will defer to JDK.
+                // * >  0  - Don't use cleaner. This will limit Netty's total direct memory
+                //           (note: that JDK's direct memory limit is independent of this).
+                long maxDirectMemory = MaxDirectMemorySetting.get();
+                if (maxDirectMemory < 0) {
+                    maxDirectMemory = SystemPropertyUtil.getLong(JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY, -1);
+                } else {
+                    logger.debug("max direct memory was defined using a setter method from the class '{}'",
+                        MaxDirectMemorySetting.class.getName());
+                }
+
+                if (maxDirectMemory < 0) {
+                    maxDirectMemory = maxDirectMemory0;
+                }
+
+                if (maxDirectMemory == 0 || !hasUnsafe() || !PlatformDependent0.hasDirectBufferNoCleanerConstructor()) {
+                    useDirectBufferNoCleaner = false;
+                    directMemoryCounter = null;
+                } else {
+                    useDirectBufferNoCleaner = true;
+                    if (maxDirectMemory < 0) {
+                        if (maxDirectMemory <= 0) {
+                            directMemoryCounter = null;
```

Comment:
Should `useDirectBufferNoCleaner = false` in this case since there is no direct memory?